app/src/main/java/org/p2p/wallet/history/repository/remote/TransactionDetailsRpcRepository.kt

```diff
@@ -1,40 +1,82 @@
 import org.p2p.solanaj.model.types.RpcRequest
-import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
+import org.p2p.wallet.history.interactor.mapper.SolanaInstructionParser
+import org.p2p.wallet.history.model.RpcTransactionSignature
 import org.p2p.wallet.rpc.RpcConstants
 import org.p2p.wallet.rpc.api.RpcHistoryApi
-import org.p2p.wallet.user.repository.UserLocalRepository
+import org.p2p.wallet.user.interactor.UserInteractor
+import timber.log.Timber
 
 class TransactionDetailsRpcRepository(
     private val rpcApi: RpcHistoryApi,
-    private val dispatchers: CoroutineDispatchers,
-    private val transactionDetailsNetworkMapper: TransactionDetailsNetworkMapper,
-    private val userLocalRepository: UserLocalRepository
+    private val userInteractor: UserInteractor,
 ) : TransactionDetailsRemoteRepository {
 
-    override suspend fun getTransactions(signatures: List<String>): List<TransactionDetails> {
-        val requestsBatch = signatures.map { signature ->
-            val encoding = buildMap {
-                this[RpcConstants.REQUEST_PARAMETER_KEY_ENCODING] =
-                    RpcConstants.REQUEST_PARAMETER_VALUE_JSON_PARSED
-            }
-            val params = listOf(signature, encoding)
-
+    override suspend fun getTransactions(
+        userPublicKey: String,
+        transactionSignature: List<RpcTransactionSignature>
+    ): List<TransactionDetails> {
+        val encoding = buildMap {
+            this[RpcConstants.REQUEST_PARAMETER_KEY_ENCODING] =
+                RpcConstants.REQUEST_PARAMETER_VALUE_JSON_PARSED
+            this[RpcConstants.REQUEST_PARAMETER_KEY_COMMITMENT] =
+                RpcConstants.REQUEST_PARAMETER_VALUE_CONFIRMED
+        }
+        val requestsBatch = transactionSignature.map { signature ->
+            val params = listOf(signature.signature, encoding)
             RpcRequest(method = RpcConstants.REQUEST_METHOD_VALUE_GET_CONFIRMED_TRANSACTIONS, params = params)
         }
 
-        return rpcApi.getConfirmedTransactions(requestsBatch)
-            .map { it.result }
-            .let {
-                withContext(dispatchers.io) {
-                    transactionDetailsNetworkMapper.fromNetworkToDomain(it) { mintAddress ->
-                        userLocalRepository.findTokenData(mintAddress)?.mintAddress.orEmpty()
+        val transactions = rpcApi.getConfirmedTransactions(requestsBatch).map { it.result }
+
+        return fromNetworkToDomain(userPublicKey, transactions).onEach { transactionDetails ->
+            transactionDetails.status =
+                transactionSignature.first { it.signature == transactionDetails.signature }.status
+        }
+    }
+
+    private fun fromNetworkToDomain(
+        tokenPublicKey: String,
+        transactions: List<ConfirmedTransactionRootResponse>
+    ): List<TransactionDetails> {
+        val transactionDetails = mutableListOf<TransactionDetails>()
+        transactions.forEach { transaction ->
+            try {
```

Comment:
it's better to extract this whole block to a method to remove indentation and complexity

Reply:
so i have already moved it to fromNetworkToDomain fun

Reply:
for this we can just create an interface and create second mock variant of interface and use it